### PR TITLE
fix lit-gpt Docker, helper.py and main.py

### PR DESCRIPTION
**Pull request by xindi-dumbledore**
Changes:
1. change in `sample-submissions/lit-gpt/helper.py` `logits = model(x, max_seq_length, input_pos)` to `logits = model(x, input_pos)`
2. remove [this line](https://github.com/llm-efficiency-challenge/neurips_llm_efficiency_challenge/blob/bbf2f8013551ee858b03feb8e97d2ef1530e924d/sample-submissions/lit-gpt/main.py#L91C24-L91C24) 

Other changes within `lit-gpt` to make it work that is not included in this PR (because it's on `lit-gpt` side)
1. uncomment all the optional requirements in `requirements.txt`
1. add `self.set_kv_cache(batch_size=1, device="cuda:0", dtype=torch.bfloat16)` in line 82 here: https://github.com/Lightning-AI/lit-gpt/blob/main/lit_gpt/model.py#L82

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]

**Comment by msaroufim** [comment]
Sorry I had to revert after noticing this will also require changes to `lit-gpt` cc @carmocca
